packages/components/src/components/legends/as-legend-size-continuous-line/as-legend-size-continuous-line.tsx

```diff
@@ -0,0 +1,210 @@
+import { Component, Prop, State } from '@stencil/core';
+
+const TEXT_MARGIN = 4;
+const MIN_LINE_SIZE = 4;
+
+@Component({
+  shadow: false,
+  styleUrl: 'as-legend-size-continuous-line.scss',
+  tag: 'as-legend-size-continuous-line'
+})
+export class LegendSizeContinuousLine {
+  @Prop() public data: LegendData[];
+  @Prop() public orientation: 'horizontal' | 'vertical' = 'vertical';
+  @Prop() public size: number = 300;
+  @Prop() public leadingLineStrokeWidth: number = 0.5;
+  // Line height of the font, which is 12 with the original as-caption
+  @Prop() public textLineHeight: number = 12;
+
+  @Prop() public xMarginFactor: number = 0.1;
+  @Prop() public yMarginFactor: number = 0.1;
+
+  @Prop() public width: number = null;
+
+  @State() private rSize = 0;
+
+
+  public componentWillUpdate() {
+    this.parseSize();
+  }
+
+  public componentWillLoad() {
+    this.parseSize();
+  }
+
+  public render() {
+    const sortedData = this.data.slice().sort(
+      (first, second) => second.width - first.width
+    );
+
+    const max = sortedData[0].width;
+    const half = max / 2;
+
+    const X_POS = Math.max(max + MIN_LINE_SIZE, this.width);
+
+    let X_OFF = (this.width - max) / 2;
+
+    if (this.orientation === 'horizontal' || this.width === null) {
+      X_OFF = 0;
+    }
+
+    // Path is painted counterclockwise starting from bottom left point
+    const realPath = [];
+    const lines = [];
+
+    sortedData.forEach((d, i) => {
+      const factor = half * (d.width / max);
+      const yRatio = i / (sortedData.length - 1);
+
+      const top = [half + factor, this.size - (this.size * yRatio)];
+      const bottom = [half - factor, this.size - (this.size * yRatio)];
+
+      if (this.orientation === 'horizontal') {
+        top.reverse();
+        bottom.reverse();
+      }
+
+      top[0] += X_OFF;
+      bottom[0] += X_OFF;
+
+      // Insert always at current index + 1 (to account for `M0 ${this.size}`)
+      realPath.splice(i, 0, `L${top.join(' ')}`);
```

Comment:
Before checking if there's a label, I think we can also have the line 100. With capital case constants, it'd be something like:

```js
        const FACTOR = HALF * (d.width / MAX);
        const Y_RATIO = i / (sortedData.length - 1);
        const TOP_X = HALF + FACTOR;
        const BOTTOM_X = HALF - FACTOR;
        const TOP_Y = BOTTOM_Y = this.size - (this.size * Y_RATIO);

        const top = this.orientation === 'vertical'
            ? [TOP_X + X_OFFSET, TOP_Y]
            : [TOP_Y + X_OFFSET, TOP_X];

        const bottom = this.orientation === 'vertical'
            ? [BOTTOM_X + X_OFFSET, BOTTOM_Y]
            : [BOTTOM_Y + X_OFFSET, BOTTOM_X];

        path.splice(i, 0, `L${top.join(' ')}`);
        path.splice(path.length - i, 0, `L${bottom.join(' ')}`);

        // label logic
```